### PR TITLE
feat: add Explicit bucket schemas API

### LINKED ISSUE #525: Support Cloud contract too
__Proposal:__

Add cloud.json contract client so we can use all APIs

__Current behavior:__

Bucket Schemas api is not available

__Desired behavior:__

I can `from influxdb_client import InfluxDBClientCloud as InfluxDBClient` to activate all needed apis.

__Alternatives considered:__

`pip install requests`

__Use case:__

We are working in Vonage over feature that involves Explicit Bucket Schemas. We want to automate their management and we are using python influxdb client. Currently only oss.json contract is parsed and added so we have to use plain requests to maintain this and it's cumbersome.

### PR DESCRIPTION
**Pull request by bednar**
Closes #525

## Proposed Changes

1. Added possibility to use [Explicit bucket schemas API](https://docs.influxdata.com/influxdb/cloud/reference/key-concepts/data-elements/#bucket-schema)
2. Added warning if the user uses Cloud only API - `BucketSchemas`, `InvokableScripts`. The following code:
   ```python
   from influxdb_client import InfluxDBClient, BucketSchemasService

   with InfluxDBClient(url="http://localhost:8086", token="my-token", org="my-org") as client:
       bucket_schemas_api = BucketSchemasService(api_client=client.api_client)

   ```
   causes:
   ```console
   The 'BucketSchemasService' is available only on the InfluxDB Cloud.

   For more info see:
       - https://docs.influxdata.com/influxdb/cloud/organizations/buckets/bucket-schema/
       - https://docs.influxdata.com/influxdb/cloud/

   You can disable this warning by:
       import warnings
       from influxdb_client.client.warnings import CloudOnlyWarning

       warnings.simplefilter("ignore", CloudOnlyWarning)
   ```
3. Addad an example: [How to manage explicit bucket schemas to enforce column names, tags, fields, and data types for your data](https://github.com/influxdata/influxdb-client-python/blob/explicit_bucket_schemas/examples/bucket_schemas.py)

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `pytest tests` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: examples/bucket_schemas.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+"""
+This example is related to `InfluxDB Cloud <https://docs.influxdata.com/influxdb/cloud/>`_ and not available
+on a local InfluxDB OSS instance.
```
Comment (powersj) [inline examples/bucket_schemas.py:3]: Do we know if this is in or not in Enterprise as well?

Comment (bednar) [reply examples/bucket_schemas.py:3]: According to the documentation, this is not available in InfluxDB Enterprise.

- https://docs.influxdata.com/enterprise_influxdb/v1.10/tools/api/

File: influxdb_client/client/warnings.py [hunk 2]
```diff
@@ -29,3 +29,24 @@ def print_warning(query: str):
...
+    - {doc_url}
+    - https://docs.influxdata.com/influxdb/cloud/
+
+You can disable this warning by:
+    import warnings
+    from influxdb_client.client.warnings import CloudOnlyWarning
+
+    warnings.simplefilter("ignore", CloudOnlyWarning)
+"""
+        warnings.warn(message, CloudOnlyWarning)
```
[the remark this answers is not in the document]
Comment (bednar) [reply influxdb_client/client/warnings.py:52]: You are right, I will add a checking to the type of instance before warning.

[the remark this answers is not in the document]
Comment (bednar) [reply influxdb_client/client/warnings.py:52]: The implementation of warning is changed. The warnings are displayed on individual actions and only if the InfluxD instance is not Cloud. For more info see:

- https://github.com/influxdata/influxdb-client-python/blob/e705ee9cad48651433e7a031d121583bebea02e9/influxdb_client/service/_base_service.py#L28
- https://github.com/influxdata/influxdb-client-python/blob/e705ee9cad48651433e7a031d121583bebea02e9/influxdb_client/service/bucket_schemas_service.py#L75

**Review by powersj — APPROVED** [review]
Thank you for fixing up the warning. I haven't gone through all the API generated code, only the logic and warnings changes that were added.

**Comment by bednar** [comment]
@powersj thanks for review. The generated code mainly contains only initialization of the base class.

**Pull request merged**